dll/win32/winmm/time.c

```diff
@@ -422,7 +423,15 @@ MMRESULT WINAPI timeGetDevCaps(LPTIMECAPS lpCaps, UINT wSize)
 MMRESULT WINAPI timeBeginPeriod(UINT wPeriod)
 {
     if (wPeriod < MMSYSTIME_MININTERVAL || wPeriod > MMSYSTIME_MAXINTERVAL)
-	return TIMERR_NOCANDO;
+		return TIMERR_NOCANDO;
+
+	/*  High resolution timer requested, use QPC */
+	if (wPeriod <= 5 && TIME_qpcFreq.QuadPart == 0) {
```

Comment:
Why 5 and/or what happens between 2 and 4?

Reply:
Because the default precision on NT is 5 milliseconds or more according to the documentation for timeGetTime, meaning any time an application requests a timer lower than that it should switch to the high-precision timer.